cpp/src/arrow/flight/sql/odbc/odbc_impl/flight_sql_statement.cc

```diff
@@ -97,7 +102,7 @@ boost::optional<Statement::Attribute> FlightSqlStatement::GetAttribute(
 
 boost::optional<std::shared_ptr<ResultSetMetadata>> FlightSqlStatement::Prepare(
```

Comment:
Working on removing `boost` as part of https://github.com/apache/arrow/pull/48067